src/VehicleSelector/VehicleSelector.tsx

```diff
@@ -0,0 +1,247 @@
+  const allProjects = uniqueSortedArray(
+    variants.map(vehicle => vehicle.projectCode)
+  );
+
+  // derive state for selected model year
+  const selectedModelYears = uniqueSortedArray(
+    value.map(vehicle => vehicle.modelYear)
+  );
+  if (selectedModelYears.length > 1)
+    throw new Error("Multiple year is ambiguous");
+  const selectedModelYear = selectedModelYears[0] ?? null;
+
+  // derive state for all model years
+  const allModelYears = uniqueSortedArray(
+    variants
+      .filter(v => v.projectCode === selectedProject)
+      .map(vehicle => vehicle.modelYear)
+  );
+
+  // derive state for selected variants
+  const selectedVariants = uniqueSortedArray(
+    value.map(vehicle => vehicle.variant)
+  ).filter(v => v !== "");
+
+  // derive state for all variants
+  const allVariants = uniqueSortedArray(
+    variants
+      .filter(
+        v =>
+          v.projectCode === selectedProject && v.modelYear === selectedModelYear
+      )
+      .map(vehicle => vehicle.variant)
+  );
+
+  // derive state for selected gates
+  const selectedGates = uniqueSortedArray(
+    value.map(vehicle => vehicle.gate)
+  ).filter(v => v !== "");
+
+  // create the selector components for project, model year, variant and gate with single select for project and model year and multi select for variant and gate
+  return (
+    <Box
+      data-testid="vehicle-select"
+      sx={{
+        display: "flex",
+        flexDirection,
+        flexWrap,
+        gap: flexDirection === "row" ? "0 24px" : 0
+      }}
+    >
+      <Box flex="40%">
+        <Autocomplete
+          label="Project Code"
+          required
+          multiple={false}
+          options={allProjects}
+          onChange={(_event, value) => {
+            const newValue = value === null ? "" : value;
+            onChange([
+              {
+                _id: "",
+                gate: "",
+                modelYear: "",
+                projectCode: newValue,
+                variant: ""
+              }
+            ]);
+          }}
+          value={selectedProject === "" ? null : selectedProject}
+        />
+      </Box>
+      <Box flex="40%">
+        <Autocomplete
+          disabled={selectedProject === null || selectedProject === ""}
+          label="Model Year"
+          required
+          multiple={false}
+          options={allModelYears}
+          onChange={(_event, value) => {
+            const newValue = value === null ? "" : value;
+            onChange([
+              {
+                _id: "",
+                gate: "",
+                modelYear: newValue,
+                projectCode: selectedProject,
+                variant: ""
+              }
+            ]);
+          }}
+          value={selectedModelYear === "" ? null : selectedModelYear}
+        />
+      </Box>
+      <Box flex="40%">
+        <Autocomplete
+          disabled={selectedModelYear === null || selectedModelYear === ""}
+          label="Vehicle Variant"
```

Comment:
I am not sure what part causes this, but there are some issues with the "Vehicle Variant" field, when field values are deleted, see the details below.

When deleting the content of the "Project code" field, every other filed should be emptied. When deleting the content of the "Model Year" filed, every field other than "Project Code" should also be deleted. (As seen in component "VehicleSelect")
However, when these two actions are done, the "Vehicle Variant" field still retains its value (albeit not editable anymore). Which can lead to a scenario, when after deleting e.g. a "Project Code", then adding a new "Project Code", an invalid "Vehicle Variant" will be selected, since it retained its value after deleting the first "Project Code".

Another issue arises when Gates are enabled, and the content of the "Vehicle Variant" is deleted. In this case, the contents of "Gates" should be emptied. But the "Gates" field retains its value in this scenario.